thrust/thrust/iterator/zip_iterator.h

```diff
@@ -308,17 +324,17 @@ make_zip_iterator(_CUDA_VSTD::tuple<Iterators...> t)
   return zip_iterator<_CUDA_VSTD::tuple<Iterators...>>(t);
 }
 
-//! \p make_zip_iterator creates a \p zip_iterator from
-//! iterators.
+//! \p make_zip_iterator creates a \p zip_iterator from iterators.
 //!
 //! \param its The iterators to copy.
 //! \return A newly created \p zip_iterator which zips the iterators.
 //!
 //! \see zip_iterator
-template <typename... Iterators>
+template <typename... Iterators,
+          ::cuda::std::enable_if_t<(::cuda::std::input_or_output_iterator<Iterators> && ...), int> = 0>
```

Comment:
Should this also use forwarding?

Reply:
I forward in the body now. Or do you mean something else?